plugin/storage/es/spanstore/service_operation.go

```diff
@@ -127,9 +171,75 @@ func getOperationsAggregation(maxDocCount int) elastic.Query {
 		Size(maxDocCount) // ES deprecated size omission for aggregating all. https://github.com/elastic/elasticsearch/issues/18838
 }
 
+func bucketOfOperationNamesToOperationsArray(buckets []*elastic.AggregationBucketKeyItem, kind string) ([]spanstore.Operation, error) {
+	result := make([]spanstore.Operation, len(buckets))
+	for i, keyItem := range buckets {
+		str, ok := keyItem.Key.(string)
+		if !ok {
+			return nil, errors.New("non-string key found in aggregation")
+		}
+		result[i] = spanstore.Operation{
+			Name:     str,
+			SpanKind: kind,
+		}
+	}
+	return result, nil
+}
+
+func bucketOfOperationsToOperationsArray(searchResult *elastic.AggregationBucketFilters) ([]spanstore.Operation, error) {
+	var result []spanstore.Operation
```

Comment:
This is concerning because we are unaware of the total length of result so not sure whether we should initialize this way or not. The best possible solution which could think of is as number of kinds are limited we can iterate through each kind fetch the total doc count of each kind, then finally create the array of total length.